File: Archive#08-801-900/844-Backspace String Compare/leetcode844.py
class Solution:
    def backspaceCompare(self, S: str, T: str) -> bool:
        def remake(s: str) -> str:
            ret = []
            for ch in s: ret = ret + [ch] if ch != '#' else ret[:-1]
            return ret
        return remake(S) == remake(T)

File: Archive#08-801-900/844-Backspace String Compare/leetcode844.rb
# @param {String} s
# @param {String} t
# @return {Boolean}
def backspace_compare(s, t)
  def remake(s)
    ret = []
    s.each_char { |ch| ret = ch == '#' ? ret[0...-1] : ret + [ch] }
    ret
  end
  remake(s) == remake(t)
end